src/thor/matrix_action.cc

```diff
@@ -42,10 +43,17 @@ std::string thor_worker_t::matrix(Api& request) {
                                       mode, max_matrix_distance.find(costing)->second);
   };
   auto timedistancematrix = [&]() {
-    return time_distance_matrix_.SourceToTarget(options.sources(), options.targets(), *reader,
-                                                mode_costing, mode,
-                                                max_matrix_distance.find(costing)->second,
-                                                options.matrix_locations());
+    time_distances =
+        forward_search
+            ? time_dist_matrix_forward_.SourceToTarget(options.sources(), options.targets(), *reader,
```

Comment:
hmm.. just realized that this sucks a bit that the caller has to know which version to request.. I'll re-write a bit